quil-rs/src/instruction/gate.rs

```diff
@@ -128,13 +131,67 @@ impl fmt::Display for GateModifier {
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq, strum::Display, strum::EnumString)]
+#[strum(serialize_all = "UPPERCASE")]
+pub enum PauliGate {
+    I,
+    X,
+    Y,
+    Z,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct PauliTerm {
+    pub word: Vec<PauliGate>,
+    pub expression: Expression,
+    pub arguments: Vec<String>,
```

Comment:
Does each word correspond to an argument? Would `Vec<(PauliGate, String)>` make sense here?

Reply:
I tried that initially, and I do like that it fits the data semantically, but the ergonomics around it are a little worse because it doesn't match the syntax as well. You have to zip it on parse, and split it on Display. On top of that tuples don't really play nice with the `rigetti_pyo3` macros, so it felt like a lot of trouble to go through for little tangible benefit. That said, I could be convinced otherwise.